miner/worker.go

```diff
@@ -1105,7 +1105,11 @@ func (w *worker) generateWork(genParams *generateParams) *newPayloadResult {
 	}
 	defer work.discard()
 	if work.gasPool == nil {
-		work.gasPool = new(core.GasPool).AddGas(work.header.GasLimit)
+		gasLimit := w.config.EffectiveGasCeil
+		if gasLimit == 0 || gasLimit > work.header.GasLimit {
+			gasLimit = work.header.GasLimit
```

Comment:
Question: If user set a more higher value in `w.config.EffectiveGasCeil`, then in line 1110, the gasLimit will set back to the old value from header.
For example, if user set 60M for this parameter, while the value in header is 30M, after line 1109 and 1110, the gasLimit is still 30M. 
Is this what we expect?

Reply:
yes, this was my intention, to always respect the 1559-configured gaslimit as an upperbound.

Reply:
emmm? so `EffectiveGasCeil` is useless if set more then block gas limit? So how do you use this value? For example, if the block limit is 30M, then how to set this?

Reply:
I'm not going to hand-hold you through this sorry.